src/devices/machine/ibm72x7377.cpp

```diff
@@ -0,0 +1,444 @@
+// license:BSD-3-Clause
+// copyright-holders:Katherine Rohl
+/*************************************************************
+
+  IBM 72X7377
+  PS/2 Type 1 DMA Controller
+
+**************************************************************/
+
+#include "emu.h"
+#include "ibm72x7377.h"
+
+//#define VERBOSE 1
+#include "logmacro.h"
+
+#ifdef _MSC_VER
+#define FUNCNAME __func__
+#else
+#define FUNCNAME __PRETTY_FUNCTION__
+#endif
+
+DEFINE_DEVICE_TYPE(IBM72X7377, ibm72x7377_device, "ibm72x7377", "IBM 72X7377 PS/2 DMA Controller")
+
+ibm72x7377_device::ibm72x7377_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: device_t(mconfig, IBM72X7377, tag, owner, clock),
+    m_maincpu(*this, ":maincpu"),
+    m_mcabus(*this, ":mb:mcabus"),
+    m_dma8237_1(*this, "dma8237_1"),
+	m_dma8237_2(*this, "dma8237_2")
```

Comment:
Using absolute device tags is a really bad idea.  As far as I can tell, all you need configured is the DMA address space, and a few write line outputs.  It doesn’t need to even know that there’s a CPU and MCA bus involved.